src/netex-convertor/single-ticket/singleTicketNetexGenerator.ts

```diff
@@ -141,53 +229,118 @@ const singleTicketNetexGenerator = (
+                [...matchingData.fareZones],
+                lineIdName,
+                'c',
+                'single'
+            );
+            fareTableFareFrameToUpdate.fareTables.FareTable.rows.FareTableRow = getFareTableElements(
+                [...matchingData.fareZones].reverse(),
+                lineIdName,
+                'r',
+                'single'
+            );
+            fareTableFareFrameToUpdate.fareTables.FareTable.includes.FareTable = getFareTables(
+                [...matchingData.fareZones].slice(0, -1),
+                lineIdName,
+                'single'
+            );
+        } else if (isReturnTicket(matchingData)) {
+            fareTableFareFrameToUpdate.priceGroups.PriceGroup = getPriceGroups(matchingData.outboundFareZones);
+            fareTableFareFrameToUpdate.fareTables.FareTable.Name.$t = matchingData.lineName;
+            fareTableFareFrameToUpdate.fareTables.FareTable.Description.$t = matchingData.serviceDescription;
+            fareTableFareFrameToUpdate.fareTables.FareTable.pricesFor.PreassignedFareProductRef.ref = "Trip@return";
+            fareTableFareFrameToUpdate.fareTables.FareTable.id = `Trip@return-SOP@p-ticket@${lineIdName}@adult`;
+            fareTableFareFrameToUpdate.fareTables.FareTable.pricesFor.SalesOfferPackageRef.ref = "Trip@return-SOP@p-ticket";
+            fareTableFareFrameToUpdate.fareTables.FareTable.usedIn.TariffRef.ref = `Tariff@return@${lineIdName}`;
+
+            fareTableFareFrameToUpdate.fareTables.FareTable.columns.FareTableColumn = getFareTableElements(
+                [...matchingData.outboundFareZones],
+                lineIdName,
+                'c',
+                'return'
+            );
+            fareTableFareFrameToUpdate.fareTables.FareTable.rows.FareTableRow = getFareTableElements(
+                [...matchingData.outboundFareZones].reverse(),
+                lineIdName,
+                'r',
+                'return'
+            );
+            fareTableFareFrameToUpdate.fareTables.FareTable.includes.FareTable = getFareTables(
+                [...matchingData.outboundFareZones].slice(0, -1),
+                lineIdName,
+                'return'
+            );
+
+        }
+
         fareTableFareFrameToUpdate.fareTables.FareTable.specifics.LineRef.ref = lineIdName;
-        fareTableFareFrameToUpdate.fareTables.FareTable.columns.FareTableColumn = getFareTableElements(
-            [...matchingData.fareZones],
-            lineIdName,
-            'c',
-        );
-        fareTableFareFrameToUpdate.fareTables.FareTable.rows.FareTableRow = getFareTableElements(
-            [...matchingData.fareZones].reverse(),
-            lineIdName,
-            'r',
-        );
-        fareTableFareFrameToUpdate.fareTables.FareTable.includes.FareTable = getFareTables(
-            [...matchingData.fareZones].slice(0, -1),
-            lineIdName,
-        );
 
         return fareTableFareFrameToUpdate;
     };
 
```

Comment:
Is there any reason that we can't have the generate function as it was? I can't see any separation of logic using the new if block

Reply:
Good spot